pkg/controllers/user/helm/common.go

```diff
@@ -142,8 +146,12 @@ func (l *Lifecycle) generateTemplates(obj *v3.App) (string, string, *common.Helm
 	if err != nil {
 		return "", "", nil, err
 	}
-
-	commands := append([]string{"template", appDir, "--name", obj.Name, "--namespace", obj.Spec.TargetNamespace}, setValues...)
+	if obj.Spec.HelmVersion != "helm_v3" {
+		helmName = "rancher-helm"
```

Comment:
This will change the value of the global variable helmName and use "rancher-helm" henceforth. We should use a local scoped variable to decide the helm version for the current obj.Spec